projects/observability/src/shared/services/csv-downloader/csv-downloader.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { Dictionary } from '@hypertrace/common';
+import { CsvDownloadFileConfig, FileDownloadService } from '@hypertrace/components';
+import { Observable, of } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class CSVDownloaderService {
+  private readonly componentHandlers: Map<string, Observable<Dictionary<unknown>[]>> = new Map();
+
+  public constructor(private readonly fileDownloaderService: FileDownloadService) {}
+
+  public downloadCsv(key: string, fileDownloadConfig?: CsvDownloadFileConfig): void {
+    this.fileDownloaderService.downloadAsCsv({
+      ...fileDownloadConfig,
```

Comment:
Should we spread this object, even if it is undefined? 🤔

Reply:
IMO should be mandatory, so updated